src/main/java/seedu/address/model/person/Session.java

```diff
@@ -86,7 +88,22 @@ public boolean equals(Object other) {
 
     @Override
     public int compareTo(Session other) {
-        return this.time.compareTo(other.time);
+        LocalDateTime thisTime = this.time;
+        LocalDateTime otherTime = other.time;
+        if (this.time.getDayOfWeek().compareTo(other.time.getDayOfWeek()) < 0) {
+            return -1;
+        } else if (this.time.getDayOfWeek().compareTo(other.time.getDayOfWeek()) == 0) {
+            int comp = thisTime.getHour() - otherTime.getHour();
+            if (comp < 0) {
+                return -1;
+            }
+            if (comp == 0) {
+                return thisTime.getMinute() - otherTime.getMinute();
+            }
+            return 1;
+        } else {
+            return 1;
+        }
```

Comment:
probably could be simplified. check if the DayOfWeek compareTo != 0 and if it is then just return the compareTo value (they are different days).

if they are the same day (compareTo = 0) then you can use toLocalTime's to convert them to a LocalTime and return that compareTo.

this is the reason I suggested splitting them into DayOfWeek and LocalTime btw

Reply:
hi ms joanna, always an honour to get your feedback 💯 ur right about the simplification. lovely will refactor.